Lenovo Yoga Tab 3 10 LTE initial support

exkc opened the pull request.

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,297 @@
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led1 {
+			function = LED_FUNCTION_HEARTBEAT;
+			color = <LED_COLOR_ID_BLUE>;
+			gpios = <&tlmm 92 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
```
what is this led? does it really exist? why is it heartbeat?

This looks like touchscreen regulator, please model it this way.

exkc replied:
> This looks like touchscreen regulator, please model it this way.

it is .i didnt know how to use regulator so um i used gpio for development/testing

>  why is it heartbeat?

yeah  I literally just copied that from somewhere else and didn't bother to change it since it is only for Development so it doesn't matter.

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,297 @@
+			summit,chip-temperature-threshold-celsius = <130>;
+			summit,usb-current-limit-microamp = <1800000>;
+
+			monitored-battery = <&bat>;
+		};
+
+};
```
indentation is broken

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,297 @@
+};
+
+//FIXME : the current developer's unit lack of a real  fuel gauge
+// therefore use the  fuel gauge from pmim as of now
```
```suggestion
// therefore use the  fuel gauge from pmic as of now
```

and only use c style comments in the dts please

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,297 @@
+};
+
+&sdhc_2 {
+	non-removable;
```
are you sure? No card detect?

exkc replied:
that is the emmc

wonderfulShrineMaidenOfParadise replied:
```
		sdhci@07864000 {
			...
			cd-gpios = <0x8d 0x26 0x01>;
			...
		};
```

TravMurav replied:
sdhc1 is emmc, sdhc2 is microsd slot.

exkc replied:
i swaped

TravMurav reviewed (commented):
For the final commit you will also need to write a suitable commit message. Please see how other 8909 devices added in our tree are described.

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,332 @@
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-up;
+	};
+	touchscreen_default: touchscreen-default-state {
```
```suggestion
	};

	touchscreen_default: touchscreen-default-state {
```

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,332 @@
+			function = "gpio";
+			drive-strength = <2>;
+			bias-disable;
+		};
+		touchscreen-pins {
```
```suggestion
		};

		touchscreen-pins {
```

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,332 @@
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+	};
+	ts_vdd_default: ts_vdd_default-state {
```
```suggestion
	};

	ts_vdd_default: ts_vdd_default-state {
```

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,332 @@
+};
+
+&blsp_i2c1 {
+	status = "okay";
+	led-controller@68 {
```
```suggestion
	status = "okay";

	led-controller@68 {
```

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,332 @@
+		};
+	};
+
+	reg_ts_vdd:regulator-gpio{
```
```suggestion
	reg_ts_vdd: regulator-gpio{
```

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,381 @@
+
+	usb_id: usb-id {
+		compatible = "linux,extcon-usb-gpio";
+		id-gpios = <&tlmm 23 GPIO_ACTIVE_HIGH>;
```
```suggestion
		id-gpios = <&tlmm 0 GPIO_ACTIVE_HIGH>;
```

```
qcom,usbid-switch = <0x8d 0x00 0x00>;
```

exkc commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,334 @@
+	status = "okay";
+};
+
+/* FIXME:  Set up SMB35 properly for the usb */
```
should I include the not working SMB35 node

TravMurav replied:
if you can't test and confirm it works properly, no. Especially for a charger chip I'd prefer changes to be tested.

exkc replied:
alr i will leave it as it

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+		};
+	};
+
+	reg_ts_vdd:regulator-gpio{
```
```suggestion
	reg_ts_vdd: regulator-touch {
```

so the name is more descriptive in case you have more than one.

exkc replied:
why touch?it should be ts

TravMurav replied:
or `ts`, at your discretion, Just not "gpio" please

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+
+		led@1 {
+			reg = <1>;
+			label = "power:led";
```
I believe this is a deprecated binding

TravMurav replied:
https://github.com/msm8916-mainline/linux/blob/wip/msm8916/6.10/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts#L158-L162

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+
+&usb_hs_phy {
+	extcon = <&pm8909_usbin>;
+	status = "disable";
```
why?

wonderfulShrineMaidenOfParadise replied:
```suggestion
	status = "disabled";
```

It's either `okay` or `disabled`.

exkc replied:
> why?

me being dumb

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+
+&usb_hs_phy {
+	extcon = <&pm8909_usbin>;
+	status = "disable";
```
```suggestion
	status = "disable";
```

It's either `okay` or `disabled`.

exkc commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+	};
+
+	bat: battery {
+		compatible = "simple-battery";
```
not sure is it a good idea?
i copied it from another device without changes

TravMurav replied:
for vm_bms there would've been a table in your downstream but if its temporary, it's not a big deal to use another device's table. The vm_bms percision is not too good for this to matter.

exkc replied:
i looked up the downsteam one look a bit smailer to that so it is fine

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+
+&sdhc_1 {
+	status = "okay";
+	non-removable;
```
```suggestion
```

https://github.com/msm8916-mainline/linux/blob/wip/msm8916/6.10/arch/arm/boot/dts/qcom/qcom-msm8909.dtsi#L950

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+
+&usb_hs_phy {
+	extcon = <&pm8909_usbin>;
+	status = "okay";
```
```suggestion
```

Not `disabled` by default.

https://github.com/msm8916-mainline/linux/blob/wip/msm8916/6.10/arch/arm/boot/dts/qcom/qcom-msm8909.dtsi#L1244-L1254

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+		compatible = "hideep,hideep-ts";
+		reg = <0x6c>;
+
+		interrupt-parent = <&tlmm>;
+		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
```
```suggestion
		interrupts-extended = <&tlmm 13 IRQ_TYPE_EDGE_FALLING>;
```

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+
+/* FIXME: Missing the real fuel gauge's node */
+&pm8909_bms {
+	power-supplies = <&pm8909_charger>;
```
```suggestion
```

No need for it if you are not using `&pm8909_charger`.

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/pinctrl/qcom,pmic-mpp.h>
+#include <dt-bindings/power/summit,smb347-charger.h>
```
```suggestion
```

Unused for now.

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,336 @@
+
+		led@1 {
+			reg = <1>;
+			function = LED_FUNCTION_POWER;
```
What is the `color`?

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,330 @@
+		};
+	};
+
+	reg_ts_vdd: regulator-ts{
```
```suggestion
	reg_ts_vdd: regulator-ts {
```

TravMurav reviewed (commented):
Small nit, otherwise this looks fine. Are you planning to work more on this devicetree?

wonderfulShrineMaidenOfParadise commented:
6.11 should just work, if you can confirm that.

exkc commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,330 @@
+
+		led@1 {
+			reg = <1>;
+			function = LED_FUNCTION_POWER;
```
i kind of want to config it with multiple functions but anyway

exkc commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,330 @@
+			reg = <1>;
+			function = LED_FUNCTION_POWER;
+			color = <LED_COLOR_ID_WHITE>;
+			led-max-microamp = <10000>;
```
From downsteam:
```
lenovo-lxf-p5100:~$ cat /sys/class/leds/red/led_max_current 
max current: 5mA
lenovo-lxf-p5100:~$ 
```


10000microamp=10mA

Ah it is missconfiged dram
AH I didn't  check it and blindly copied from the example code but  led didnt burn humm.

exkc commented:
> Small nit, otherwise this looks fine. Are you planning to work more on this devicetree?

I think that's it.the only thing left is the usb  and the other things that mainline don't have driver for

as you know I was quite struggling to get the USB work (power out/mode switching/guest mode)

exkc commented:
Oh wait let me frix a small typo

exkc commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,359 @@
+&blsp_i2c1 {
+	status = "okay";
+	
+	/* TODO: add Bochs BMA223 accelerometer/ELAN EPL2102 light sensor here */
```
Bochs BMA223 accelerometer: lack of mainline driver
ELAN EPL2102 light sensor: lack of mainline driver,rare chip

exkc replied:
Also TS3A225E(Jack detention IC) as well but I'm not going to put TS3A225E in the comment bc 1. audio don't work yet 2.  It seem  the mainline don't want to support Jack detention IC (See https://github.com/msm8916-mainline/linux/commit/26f59e3d9f59554c1cf826786201838cb7146339)

wonderfulShrineMaidenOfParadise replied:
> Bochs BMA223 accelerometer: lack of mainline driver

It could have been driven by `bmc150-accel`, I assume?
https://github.com/msm8916-mainline/linux/blob/wip/msm8916/6.10/drivers/iio/accel/bmc150-accel-i2c.c#L264-L273

exkc replied:
it failed to read chip id from the chip

i could try to remove the probing code of it to force it work on the chip but i havent try that.Also that is a hack.

btw i have done smailer thing to UDL

TravMurav commented:
This looks good, however few things still left:

- Please fix checkpatch errors (for author matching signed-off, you can do `commit --amend --author "your name <your@email>"` or something like that)
- Please rebase on the latest branch (currently it's `wip/msm8916/6.12-rc4`) and check that there are no regressions

exkc commented:
oh wait i forgot the hall sensor
 but I don't know how to test it and idk will gpio-hell work on it
anyway i will add it to the tree using gpio-hell

Commit:
ARM: dts: qcom: lenovo-lxf-p5100: Add initial device tree

Lenovo Yoga Tab 3 10 LTE is a tablet using the MSM8909 SoC released in 2016.

Add a device tree for with initial support for:

- GPIO keys
- pm8909-vibrator
- SDHCI (internal and external storage)
- UART
- WCNSS (WiFi/BT)
- Regulators
- Touchscreen
- PWM panel backlight
- Hall
- Power LED
- Framebuffer
- USB (Dont Work)
- Magnetometer

Signed-off-by: Karl Chan <exxxxkc@getgoogleoff.me>

exkc commented:
that really should be the final I guess.I will do a boot test

exkc commented:
![image](https://github.com/user-attachments/assets/72f56165-04a5-4211-a3c7-5ec3e54b9236)
it boot

exkc commented:
closed bc of missclick

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,370 @@
+			linux,code = <KEY_VOLUMEUP>;
+		};
+
+		hall-sensor {
```
we usually separate the hall sensor to it's own device, also these names doesn't conform to the dt bindings (should start with button- or event-)

https://github.com/msm8916-mainline/linux/blob/wip/msm8916/6.10/arch/arm64/boot/dts/qcom/msm8916-samsung-a2015-common.dtsi#L77

would you mind checking your dts with dtbs_check? I think it's `make CHECK_DTBS=y W=1 qcom-msm8909-lenovo-lxf-p5100.dtb`

exkc replied:
```
 make ARCH=arm O=build/  CHECK_DTBS=y W=1 qcom-msm8909-lenovo-lxf-p5100.dtb
make[1]: Entering directory '/tmp/linux/build'
make[3]: *** No rule to make target 'arch/arm/boot/dts/qcom-msm8909-lenovo-lxf-p5100.dtb'.  Stop.
make[2]: *** [/tmp/linux/Makefile:1399: qcom-msm8909-lenovo-lxf-p5100.dtb] Error 2
make[1]: *** [/tmp/linux/Makefile:224: __sub-make] Error 2
make[1]: Leaving directory '/tmp/linux/build'
make: *** [Makefile:224: __sub-make] Error 2

```
eh?

TravMurav replied:
perhaps add qcom/ to the dtb path, tho I thought it still uses flat scheme for arm32 even after the recent sorting refactor

exkc replied:
[a.log](https://github.com/user-attachments/files/17575916/a.log)
i run it at the latest changes

TravMurav replied:
do you have the latest version of dt-schema? 
`pip3 install git+https://github.com/devicetree-org/dt-schema.git@master`

exkc replied:
i did `paru -S python-dtschema-git` not long b4 i run that

TravMurav replied:
cache-controller is in the recent dt-schema, I'm not sure why check crashes, it shouldn't...

TravMurav replied:
regardless of the bindings check, the node name here is still wrong.

https://elixir.bootlin.com/linux/v6.11.5/source/Documentation/devicetree/bindings/input/gpio-keys.yaml#L26

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,370 @@
+		shutdown-gpios = <&tlmm 27 GPIO_ACTIVE_HIGH>;
+
+		led@1 {
+			reg = <1>;
+			function = LED_FUNCTION_POWER;
+			led-max-microamp = <5000>;
```
I thought I saw a color prop here before? What does this led do even?

exkc replied:
I somehow revent some change during the rebase. I added those back i tink um i guess missed one

exkc replied:
> What does this led do even?

it is notification / chagrining on stock android

it is at the same place at the power button so i added this line `function = LED_FUNCTION_POWER;`

TravMurav replied:
please use LED_FUNCTION_INDICATOR or LED_FUNCTION_STATUS for notification led so software can find it

exkc replied:
https://github.com/msm8916-mainline/linux/pull/374#discussion_r1821977788

ah also it is a heatbeat led.there are so many function cramming to the one led function

also LED_FUNCTION_CHARGING is also a thing

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,370 @@
+
+		reset-gpios = <&tlmm 94 GPIO_ACTIVE_LOW>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&magnetometer_default>;
```
```suggestion
		pinctrl-0 = <&magnetometer_default>;
		pinctrl-names = "default";
```

-names always last

TravMurav reviewed (commented):
Was about to merge but saw a few more things to improve

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts.orig:
```diff
@@ -0,0 +1,383 @@
+// SPDX-License-Identifier: GPL-2.0-only
```
the .orig file doesn't belong in the commit. Please check if you checked in correct things

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,383 @@
+	};
+
+	chosen {
+		stdout-path = "serial0";
+
+		framebuffer@83200000 {
+			compatible = "simple-framebuffer";
+			reg = <0x83200000 (800 * 1280 * 3)>;
```
I think dtbs check had a valid warning here about missing #cell counts

https://elixir.bootlin.com/linux/v6.11.5/source/arch/arm/boot/dts/qcom/msm8226-motorola-falcon.dts#L19-L23

exkc replied:
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L17
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L18
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L19
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L20
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L21
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L22
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L23
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L24
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L25
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L26
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L27
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L28
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L29
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L30
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L31
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L32
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L33
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L34
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L35
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L36
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L37
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L38
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L39
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L40
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L41
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L42
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L43
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L44
https://github.com/msm8916-mainline/linux/blob/bad4374df58ffc7516b4e05a6ea47759015c3dcc/arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts#L45
it is fine? look

wonderfulShrineMaidenOfParadise replied:
Not really when a warning exists. I have no idea how to fix it yet.

TravMurav replied:
The warning says "linux is allowed to assume the address is 64bit and size is 32bit, but you have both 32bit"

Both are broken, please add explicit address/size cells as in the example I've linked

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,383 @@
+
+		label = "GPIO Buttons";
+
+		volume-up {
```
This name (and the one below) still doesn't match the binding. (Upstream instantly rejects things like that...) dtb check would've found that so I'd really like you to make that work and check your dtb with it.

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,383 @@
+		pinctrl-0 = <&magnetometer_default>;
+		pinctrl-names = "default";
+	};
+
```
```suggestion
```

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,383 @@
+		hall-pins {
+			pins = "gpio36";
+			function = "gpio";
+
```
```suggestion
```

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,383 @@
+};
+
+&tlmm {
+	gpio_keys_default: gpio-keys-default-state {
+		volume-up-pins {
+			pins = "gpio90";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+	};
```
for all the -state nodes with only one -pins node, unfold it please.

https://elixir.bootlin.com/linux/v6.11.5/source/arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts#L245-L268

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,375 @@
+		};
+	};
+
+	magnetometer@c {
```
looking at this again, 0xc is before 0x68. Please sort those per unit address when it exists

TravMurav commented on arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dts:
```diff
@@ -0,0 +1,379 @@
+};
+
+/* FIXME: Missing the real fuel gauge's node */
+&pm8909_bms {
+	status = "okay";
+};
```
```
arch/arm/boot/dts/qcom/qcom-msm8909-lenovo-lxf-p5100.dtb: vm-bms@4000: 'monitored-battery' is a required property
	from schema $id: http://devicetree.org/schemas/power/supply/qcom,pm8916-bms-vm.yaml#
```

https://github.com/msm8916-mainline/linux/blob/wip/msm8916/6.10/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts#L328-L333

wonderfulShrineMaidenOfParadise replied:
> 
```suggestion
&pm8909_bms {
	monitored-battery = <&bat>;
	status = "okay";
};
```

TravMurav commented:
This is good now, thanks! Feel free to open a follow up with extra features or fixup commits later if you think the base dts needs to be changed
